metadata.md

```diff
@@ -0,0 +1,98 @@
+# Metadata Schema - Version 1.0
+
+In order to ensure interoperability between implementations of the Arrow in-memory format
+and associated storage formats such as Parquet and Feather, it is necessary to
+include additional top-level metadata.
+
+This information includes:
+
+-   spatial properties of each geometry column
+-   name of the primary geometry column
+
+Metadata is to be stored as a JSON-encoded UTF-8 string under the "geo" key within the
+top-level metadata object of the in-memory or file format.
+
+For clarity, the following shows the unencoded JSON structure:
+
+```python
+"geo": {
+        "columns": {...},  # REQUIRED: see below
+        "creator": {
+            "library": "<library name>",  # REQUIRED
+            "version": "<library version>",  # REQUIRED
+        },
+        "primary_column": "<primary geometry column name>",  # REQUIRED
+        "version": "1.0.0"  # REQUIRED
+    }
+```
+
+Each geometry column in the dataset must be included in the `columns` field above
+with the following content, keyed by the column name:
+
+```python
+"<column_name>": {
+    "bounds": [<xmin>, <ymin>, (zmin), <xmax>, <ymax>, (zmax)],  # OPTIONAL
+    "crs": "<WKT representation of CRS>",  # REQUIRED
+    "encoding": "WKB",  # REQUIRED
+}
+```
+
+## Bounding boxes
+
+Bounding boxes are used to help define the spatial extent of each geometry column.
+Implementations of this schema may choose to use those bounding boxes to filter
+partitions (sub-datasets) of a dataset. For example, Parquet supports partitions
+as individual files ([reference](https://arrow.apache.org/docs/python/parquet.html?highlight=pyarrow%20parquet%20partition#partitioned-datasets-multiple-files)).
+
+These must be encoded with the minimum and maximum values of each dimension:
+
+-   2D: `[<xmin>, <ymin>, <xmax>, <ymax>]`
+-   3D: `[<xmin>, <ymin>, <zmin>, <xmax>, <ymax>, <zmax>]`
+
+## Coordinate Reference System (CRS) encoding
+
+CRS information must be encoded using Well-Known Text ([WKT](https://proj.org/faq.html#what-is-the-best-format-for-describing-coordinate-reference-systems).
+
+Multiple dialects of WKT exist, including:
+
+-   GDAL WKT
+-   ESRI WKT
+-   WKT2:2015 (ISO 19162:2015)
+-   WKT2:2018 (ISO 19162:2018)
+
+Any WKT encoding used in must be supported by the
+[Proj](https://proj.org/index.html) library.
```

Comment:
Are we allowing any of the dialects to be used? Does part of the WKT string define which dialect is in use? Or will Proj understand which dialect the string is?

Reply:
I think any of the dialects are fine, so long as Proj can understand them.

[Docs](https://proj.org/development/reference/functions.html#_CPPv430proj_context_guess_wkt_dialectP10PJ_CONTEXTPKc) and implementation [here](https://github.com/OSGeo/PROJ/blob/7.0.0/src/iso19111/c_api.cpp#L357-L379) and [here](https://github.com/OSGeo/PROJ/blob/7.0.0/src/iso19111/io.cpp#L6141-L6200).

Uses keywords specific to each dialect to try and guess the dialect.

Looks like WKT2:2018 is now WKT2:2019.

Reply:
I am wondering if WKT is necessarily the best option if eg an EPSG code is available (I have seen some compatibility issues with flavors of WKT, recently). Or if we should not allow the possibilty to specify EPSG codes.

Reply:
I took the comments in the original PR in geopandas to indicate that WKT was the preferred option.  I'm not opposed to EPSG codes, but we'd need to store more information in the metadata to make interpretation of CRS info clear.

Reply:
@jorisvandenbossche Do we want to add EPSG codes in this PR or consider adding it later?

Reply:
> Do we want to add EPSG codes in this PR or consider adding it later?

I am fine to consider it later / discuss in a separate issue.